app/controllers/automated_tests_controller.rb

```diff
@@ -220,4 +221,29 @@ def required_params
   def assignment_params
     params.require(:assignment).permit(*required_params)
   end
+
+  def find_or_create_test_grouping(assignment_id)
+    test_student = TestStudent.find_or_create_by(user_name: User::TEST_STUDENT_USER_NAME) do |user|
+      user.first_name = 'Test'
+      user.last_name = 'Student'
+    end
+    test_groupings = test_student.groupings
+    test_grouping = test_groupings.find_by(assessment_id: assignment_id)
+    return test_grouping unless test_grouping.nil?
+
+    if test_groupings.empty?
+      test_group = Group.new
+      test_group.group_name = test_group.get_autogenerated_group_name
+      test_group.save
+    else
+      test_group = test_groupings.first.group
+    end
+    test_grouping = test_group.groupings.create!(assessment_id: assignment_id)
+    StudentMembership.find_or_create_by(user_id: test_student.id, grouping_id: test_grouping.id) do |membership|
+      membership.membership_status = StudentMembership::STATUSES[:inviter]
+    end
+    test_grouping
+  rescue ActiveRecord::RecordInvalid, RecordNotSaved
```

Comment:
This will only catch an error if the grouping isn't created. What if the TestStudent, Group, or StudentMembership can't be created?